### ISSUE TITLE
fix(CoefficientsDisplay): If estimators have different features then `sharey` should be False

### ISSUE THREAD

**Issue by auguste-probabl** [probabl-ai/skore#2393]
### Describe the bug

When plotting coefficients of a ComparisonReport, if the estimators have different input features, the subplots should each have their own y-axis.

### Steps/Code to Reproduce

Reproducer:
```python
from sklearn.datasets import load_iris
from sklearn.pipeline import Pipeline
from sklearn.preprocessing import PolynomialFeatures
from sklearn.linear_model import LogisticRegression
from skore import ComparisonReport, CrossValidationReport

X, y = load_iris(return_X_y=True)
estimator_1 = LogisticRegression(random_state=42)
estimator_2 = Pipeline(
    [
        ("poly", PolynomialFeatures()),
        ("predictor", LogisticRegression(random_state=0)),
    ]
)

report = ComparisonReport(
    [CrossValidationReport(estimator_1, X, y), CrossValidationReport(estimator_2, X, y)]
)
display = report.inspection.coefficients()
display.plot()
```

### Expected Behavior

![Image](https://github.com/user-attachments/assets/bdc2bd85-d367-43a1-be88-399d4aa79031)

but each subplot should have its own y-axis

### Actual Behavior

![Image](https://github.com/user-attachments/assets/bdc2bd85-d367-43a1-be88-399d4aa79031)

### Environment

main (fce4f566b3ba1aeeaf94faf4fc9a490b4c199809)

**Comment by auguste-probabl** [comment]
I need to check but based on #2387 it seems that this is only the case for `ComparisonReport[CrossValidationReport]`
